src/no-dead-relative-link.js

```diff
@@ -39,21 +38,57 @@ async function validateLinkNode(linkNode, context, options) {
 }
 
 async function validateRelativeLink(linkNode, context, options) {
-    let linkAbsoutePath = path.resolve(path.dirname(context.getFilePath()), linkNode.url);
-    let linkURL = new URL("file://" + linkAbsoutePath);
-    let linkedFileExtension = path.extname(linkURL.pathname);
+    let linkURL = getLinkURL(linkNode.url, context, options);
+    if (!await fileExists(url.fileURLToPath(linkURL))) {
+        if (options["route-map"]) {
+            if (!await routedLinkExists(linkNode, context, options)) {
+                reportError(linkNode, context, `${path.basename(linkURL.pathname)} does not exist`);
+            }
+        }
+        return;
+    }
```

Comment:
If r`oute-map` option is **not configured and the target file does not exist**, we should be throwing an error by calling the `reportError` function. I think the reason why this scenario is not caught by the unit test is because you're using the same "configuration/options" for all test scenarios. If you use different combination of options, the unit test will probably catch this issue as well.

Reply:
Right, it came back to me why I had used a flag in the first place. Surprisingly it still reports an error somehow.